presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -215,7 +217,9 @@ public PrestoSparkQueryExecutionFactory(
     public IPrestoSparkQueryExecution create(
             SparkContext sparkContext,
             PrestoSparkSession prestoSparkSession,
+            Optional<String> userAgent,
```

Comment:
Let's pass it through the `PrestoSparkSessionContext` / `PrestoSparkSession`. The `SessionContext` interface has the

```
    @Nullable
    String getUserAgent();
```

declared. We just need to implement it in the `PrestoSparkSessionContext` and in the `PrestoSparkSession`

Reply:
@arhimondr : it's the other way around, `PrestoSparkSessionContext` is created in `PrestoSparkQueryExecutionFactory#create`. Thus the user agent need to passed in here (so `PrestoSparkSessionContext` will have the user agent we want